src/components/pages/document/create/uploader/tabs.tsx

```diff
@@ -3,13 +3,14 @@ import { useIsMount } from "@hooks/use-is-mount";
 import useTranslation from "next-translate/useTranslation";
 import React, { useEffect, useState } from "react";
 
+import ExternalUrl from "../resource-url";
 import useManageDocument from "./document-upload-provider";
 import DocumentDropzone from "./dropzone";
 import MyDocumentUploads from "./my-uploads";
 
-export default function DocumentUploaderTabs({ onChange }) {
+export default function DocumentUploaderTabs({ onChange, externalUrl }) {
   const { t } = useTranslation();
-  const [tabIndex, setTabIndex] = useState(0);
+  const [tabIndex, setTabIndex] = externalUrl ? useState(2) : useState(0);
```

Comment:
Please avoid putting `useState` in a condition like above please use below this applies to components as well
```
const [tabIndex, setTabIndex] = useState(externalUrl ? 2 : 0);
```